i6_models/parts/conformer/frontend.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import annotations
+
+__all__ = ["ConformerFrontendV1", "ConformerFrontendV1Config"]
+from typing import Optional
+from dataclasses import dataclass
+import torch
+from torch import nn
+
+from i6_models.config import ModelConfiguration
+
+
+@dataclass
+class ConformerFrontendV1Config(ModelConfiguration):
+    """
+    :param feature_dim: Feature dimension of the input data
+    :param hidden_dim: Hidden dimension used in the model internally
+    :param dropout: Dropout value after linear transformation
+    :param conv_stride: Stride of down-sampling cov
+    :param conv_kernel: Kernel size of down-sampling conv
+    :param conv_padding: Padding factor of down-sampling conv
+    :param spec_aug_cfg: TODO
+    """
+
+    feature_dim: int
+    hidden_dim: int
+    dropout: float
+    conv_stride: int
+    conv_kernel: int
+    conv_padding: int
+
+    # nested configurations
+    spec_aug_cfg: Optional[ModelConfiguration]  # TODO
+
+
+class ConformerFrontendV1(nn.Module):
+    """
+    Frontend part of the standard conformer doing down-sampling
+    """
+
+    def __init__(self, cfg: ConformerFrontendV1Config):
+        """
+        :param cfg: conformer frontend configuration
+        """
+        super().__init__()
+        if cfg.spec_aug_cfg:
+            # TODO: This still needs to be implemented
+            raise NotImplementedError
+        else:
+            self.spec_aug = None
+        self.subsampling = nn.Conv1d(
```

Comment:
I would rename `self.subsampling` to `self.conv` or `self.conv_subsampling` or something similar to make clear that there is a convolution involved and not only subsampling by dropping frames or so.